lib/components/normal-components/Board.ts

```diff
@@ -146,11 +146,31 @@ export class Board extends Group<typeof boardProps> {
   }
 
   doInitialPcbDesignRuleChecks() {
+    if (this.root?.pcbDisabled) return
+    if (this.getInheritedProperty("routingDisabled")) return
+    this.updatePcbDesignRuleChecks()
+  }
+
+  updatePcbDesignRuleChecks() {
     if (this.root?.pcbDisabled) return
     if (this.getInheritedProperty("routingDisabled")) return
     const { db } = this.root!
-    const errors = checkEachPcbTraceNonOverlapping(db.toArray())
-    for (const error of errors) {
+
+    // Store existing non-DRC errors
+    const existingErrors = db.pcb_trace_error
+      .list()
+      .filter((error) => !error.message.includes("overlaps with"))
+
+    // Clear all trace errors
+    db.pcb_trace_error.list().forEach((error) => {
+      db.pcb_trace_error.delete(error.pcb_trace_error_id)
```

Comment:
don't delete all pcb trace errors, only delete ones that are created from this board's DRC. You may need to store a property to indicate it was created by this Board